applications/FluidDynamicsHydraulicsApplication/python_scripts/navier_stokes_two_fluid_hydraulic_fractional_solver.py

```diff
@@ -0,0 +1,644 @@
+
+        if self.mass_source:
+            water_volume_after_transport = KratosCFD.FluidAuxiliaryUtilities.CalculateFluidNegativeVolume(self.GetComputingModelPart())
+            water_cut_volume_after_transport = KratosCFD.FluidAuxiliaryUtilities.CalculateFluidCutElementsNegativeVolume(self.GetComputingModelPart())
+            absolute_water_volume_error = water_volume_after_transport -  self.__initial_water_system_volume
+            water_cut_element_error = absolute_water_volume_error /water_cut_volume_after_transport
+            air_cut_volume_after_transport = KratosCFD.FluidAuxiliaryUtilities.CalculateFluidCutElementsPositiveVolume(self.GetComputingModelPart())
+            air_cut_element_error = absolute_water_volume_error / air_cut_volume_after_transport
+        else:
+            water_cut_element_error = 0.0
+            air_cut_element_error =0.0
+
+        self.main_model_part.ProcessInfo.SetValue(KratosCFD.WATER_VOLUME_ERROR, water_cut_element_error)
+        self.main_model_part.ProcessInfo.SetValue(KratosCFD.AIR_VOLUME_ERROR, air_cut_element_error)
+
+    def __CalculateArtificialViscosity(self):
+        properties_1 = self.main_model_part.Properties[1]
+        water_dynamic_viscosity_max = self.artificial_limiter_coefficient * properties_1.GetValue(KratosMultiphysics.DYNAMIC_VISCOSITY)
+        KratosHydraulics.HydraulicFluidAuxiliaryUtilities.CalculateNonIntersectedElementsArtificialViscosity(self.main_model_part, water_dynamic_viscosity_max)
+
+    def __PerformLevelSetConvection(self):
+        # Solve the levelset convection problem
+        self._GetLevelSetConvectionProcess().Execute()
+
+    def __PerformNSFractionalSplitting(self):
+        domain_size = self.main_model_part.ProcessInfo[KratosMultiphysics.DOMAIN_SIZE]
+        velocity_components = [KratosMultiphysics.VELOCITY_X,KratosMultiphysics.VELOCITY_Y,KratosMultiphysics.VELOCITY_Z]
+        fractional_velocity_componentes = [KratosCFD.FRACTIONAL_VELOCITY_X, KratosCFD.FRACTIONAL_VELOCITY_Y, KratosCFD.FRACTIONAL_VELOCITY_Z]
+        # Trasfer velocity node fixity to fractional velocity
+        for i in range(domain_size):
+            self.VelocityBoundaryConditionFractional(fractional_velocity_componentes[i], velocity_components[i])
+        self._GetNSFractionalSplittingProcess().Execute()
+        # Trasfer velocity slip condition to fractional velocity
+        self.__SlipConditonFractionalFixity()
+
+    def __SlipConditonFractionalFixity(self):
+        for node in self.GetComputingModelPart().Nodes:
+            if node.Is(KratosMultiphysics.SLIP):
+                n= node.GetSolutionStepValue(KratosMultiphysics.NORMAL)
+                n/= math.sqrt(n[0]**2+n[1]**2+n[2]**2)
+                v= node.GetSolutionStepValue(KratosCFD.FRACTIONAL_VELOCITY)
+                v_prooj = self.DotProduct(v,n)
+                v-= v_prooj*n
+                node.SetSolutionStepValue(KratosCFD.FRACTIONAL_VELOCITY,v)
```

Comment:
This must definitively be done in the cpp side. Do it ASAP in another PR. Furthermore, I think should be done within an `ApplyHydraulicSlipConditionProcess`.

Reply:
Thanks for the suggestion. I'll do it ASAP to avoid it here.